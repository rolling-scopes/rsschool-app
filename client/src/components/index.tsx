export * from './AdminSider';
export * from './CrossCheckComments';
export * from './Footer';
export * from './GithubAvatar';
export * from './GithubUserLink';
export * from './Header';
export * from './LoadingScreen';
export { MentorSearch } from './MentorSearch';
export * from './PageLayout';
export * from './Rating';
export * from './CommentModal';
export * from './StudentMentorModal';
export { StudentSearch } from './StudentSearch';
export { UserSearch } from './UserSearch';
export * from './withSession';
export { RegistrationPageLayout } from './RegistartionPageLayout';
