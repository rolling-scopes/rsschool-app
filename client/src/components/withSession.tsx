import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { LoadingScreen } from './LoadingScreen';
import Router from 'next/router';

export type Role = 'student' | 'mentor' | 'coursemanager';

export interface Session {
  id: number;
  githubId: string;
  isAdmin: boolean;
  isHirer: boolean;
  isActivist: boolean;
  roles: { [key: number]: Role };
  coursesRoles?: { [key: string]: ('taskOwner' | 'juryActivist' | 'manager' | 'supervisor')[] | undefined };
}

let sessionCache: Session | undefined;

function withSession(WrappedComponent: React.ComponentType<any>, requiredRole?: Role) {
  return (props: any) => {
    const [isLoading, setLoading] = useState(true);
    const [session, setSession] = useState<Session | undefined>();

    useEffect(() => {
      const load = async () => {
        if (sessionCache != null) {
          setLoading(false);
          setSession(sessionCache);
          return;
        }
        try {
          const response = await axios.get(`/api/session`);
          setLoading(false);
          sessionCache = response.data.data;
          setSession(sessionCache);
        } catch (e) {
          setLoading(false);
          Router.push('/login', {
            pathname: '/login',
            query: {
              url: encodeURIComponent(document.location.pathname + document.location.search),
            },
          });
        }
      };

      load();
    }, []);

    if (session && requiredRole) {
      const { roles, isAdmin } = session;
      if (roles[props.course.id] !== requiredRole && !isAdmin) {
        return (
          <h4 className="m-5 d-flex justify-content-center">
            You are not [{requiredRole}] in {props.course.alias}
          </h4>
        );
      }
    }
    return (
      <LoadingScreen show={isLoading}>{session && <WrappedComponent session={session} {...props} />}</LoadingScreen>
    );
  };
}

export default withSession;
export { withSession };
