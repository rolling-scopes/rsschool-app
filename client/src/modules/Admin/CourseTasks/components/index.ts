export { default as MobileCardView } from './MobileCardView';
