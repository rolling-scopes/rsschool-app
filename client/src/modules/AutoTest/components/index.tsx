export { default as TaskCard, type TaskCardProps } from './TaskCard/TaskCard';
export { default as TaskDescription } from './TaskDescription/TaskDescription';
export { default as TaskCardColumn } from './TaskCardColumn/TaskCardColumn';
export { default as TaskDeadlineDate, type TaskDeadlineDateProps } from './TaskDeadlineDate/TaskDeadlineDate';
export { default as VerificationsTable, type VerificationsTableProps } from './VerificationsTable/VerificationsTable';
export { default as Exercise } from './Exercise/Exercise';
export { default as Coding } from './Coding/Coding';
export { default as JupyterNotebook } from './JupyterNotebook/JupyterNotebook';
export { default as SelfEducation } from './SelfEducation/SelfEducation';
export { default as VerificationInformation } from './VerificationInformation/VerificationInformation';
export { default as StatusTabs } from './StatusTabs/StatusTabs';
