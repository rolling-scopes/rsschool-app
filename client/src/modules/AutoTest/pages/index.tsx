export { default as AutoTests, type AutoTestsProps } from './AutoTests/AutoTests';
export { default as Task, type AutoTestTaskProps } from './Task/Task';
