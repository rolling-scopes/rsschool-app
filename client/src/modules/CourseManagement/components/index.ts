export { SelectCourseTasks } from './SelectCourseTasks/SelectCourseTasks';
export { CertificateCriteriaModal } from './CertificateCriteriaModal/CertificateCriteriaModal';
export { ExpelCriteriaModal } from './ExpelCriteriaModal/ExpelCriteriaModal';
