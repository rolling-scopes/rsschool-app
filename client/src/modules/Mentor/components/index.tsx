export { MentorDashboard } from './MentorDashboard';
export { Notification } from './Notification';
export { Instructions } from './Instructions';
export { StudentsTable } from './StudentsTable';
