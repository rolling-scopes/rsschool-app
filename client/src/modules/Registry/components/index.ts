export { GeneralMentor } from './GeneralMentor';
export { Mentorship } from './Mentorship';
export { Info } from './Info';
