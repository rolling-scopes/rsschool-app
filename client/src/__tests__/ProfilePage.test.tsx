import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NextRouter } from 'next/router';
import { Session } from 'components/withSession';
import { ProfilePage } from '../pages/profile';

jest.mock('next/config', () => () => ({}));
jest.mock('api', () => ({
  ProfileApi: jest.fn(),
  UsersNotificationsApi: jest.fn(),
  NotificationsApi: jest.fn(),
  CoursesApi: jest.fn(),
  CoursesTasksApi: jest.fn(),
  StudentsScoreApi: jest.fn(),
}));

const profile = {
  permissionsSettings: {
    isProfileVisible: { all: true },
    isAboutVisible: { mentor: true, student: false, all: false },
    isEducationVisible: { mentor: true, student: false, all: false },
    isEnglishVisible: { student: false, all: false },
    isEmailVisible: { student: false, all: false },
    isTelegramVisible: { student: false, all: false },
    isSkypeVisible: { student: false, all: false },
    isPhoneVisible: { student: false, all: false },
    isContactsNotesVisible: { student: true, all: false },
    isLinkedInVisible: { mentor: true, student: false, all: false },
    isPublicFeedbackVisible: { mentor: true, student: true, all: false },
    isMentorStatsVisible: { mentor: true, student: true, all: false },
    isStudentStatsVisible: { student: false, all: false },
  },
  generalInfo: {
    aboutMyself: 'Test',
    educationHistory: [
      {
        graduationYear: '2019',
        faculty: 'TT',
        university: 'Test',
      },
    ],
    englishLevel: 'a2+',
    location: {
      countryName: 'Belarus',
      cityName: 'Minsk',
    },
  },
  contacts: {},
  mentorStats: [{}],
  studentStats: [
    {
      courseFullName: 'test',
      courseName: 'test',
      locationName: 'Minsk',
      tasks: [
        {
          interviewFormAnswers: {},
        },
      ],
    },
  ],
  publicFeedback: [{}],
  stageInterviewFeedback: [{}],
};
const session = {
  id: 2020,
  githubId: 'mikhama',
  isAdmin: true,
  isHirer: false,
  isActivist: false,
  courses: {
    13: {
      roles: ['manager'],
    },
    1: {
      roles: ['mentor'],
    },
    2: {
      roles: ['student'],
    },
  },
} as Session;
const router = {
  query: {
    githubId: 'petrov',
  },
  asPath: '/#edit/',
} as unknown as NextRouter;
const state = {
  profile,
  isInitialPermissionsSettingsChanged: false,
  isInitialProfileSettingsChanged: false,
};

describe('ProfilePage', () => {
  describe('Should render correctly', () => {
    it('if full profile info is in the state', () => {
      const wrapper = shallow(<ProfilePage session={session} router={router} />);
      wrapper.setState(state);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });

  const wrapper = shallow(<ProfilePage session={session} router={router} />);
  const instance: any = wrapper.instance();
  describe('onPermissionsSettingsChange', () => {
    describe('Should set state correctly', () => {
      it('if permissions for student role were changed', async () => {
        const event = {
          target: {
            checked: true,
          },
        };
        const changedPermissionsSettings = {
          permissionName: 'isEmailVisible',
          role: 'student',
        };
        wrapper.setState(state);
        await instance.onPermissionsSettingsChange(event, changedPermissionsSettings);
        expect((wrapper.state() as any).profile.permissionsSettings.isEmailVisible).toEqual({
          student: true,
          all: false,
        });
        expect((wrapper.state() as any).isInitialPermissionsSettingsChanged).toBe(true);
      });
      it('if permissions for mentor role were changed', async () => {
        const event = {
          target: {
            checked: false,
          },
        };
        const changedPermissionsSettings = {
          permissionName: 'isLinkedInVisible',
          role: 'mentor',
        };
        wrapper.setState(state);
        await instance.onPermissionsSettingsChange(event, changedPermissionsSettings);
        expect((wrapper.state() as any).profile.permissionsSettings.isLinkedInVisible).toEqual({
          mentor: false,
          student: false,
          all: false,
        });
        expect((wrapper.state() as any).isInitialPermissionsSettingsChanged).toBe(true);
      });
      it('if permissions for all roles were changed', async () => {
        const event = {
          target: {
            checked: true,
          },
        };
        const changedPermissionsSettings = {
          permissionName: 'isEducationVisible',
          role: 'all',
        };
        wrapper.setState(state);
        await instance.onPermissionsSettingsChange(event, changedPermissionsSettings);
        expect((wrapper.state() as any).profile.permissionsSettings.isEducationVisible).toEqual({
          mentor: true,
          student: true,
          all: true,
        });
        expect((wrapper.state() as any).isInitialPermissionsSettingsChanged).toBe(true);
      });
    });
  });
  describe('onProfileSettingsChange', () => {
    describe('Should set state correctly', () => {
      it('if "profile.generalInfo.location" was changed', async () => {
        const event = {
          countryName: 'USA',
          cityName: '51 zone',
        };
        const path = 'generalInfo.location';
        wrapper.setState(state);
        await instance.onProfileSettingsChange(event, path);
        expect((wrapper.state() as any).profile.generalInfo.location.countryName).toBe('USA');
        expect((wrapper.state() as any).profile.generalInfo.location.cityName).toBe('51 zone');
        expect((wrapper.state() as any).isInitialProfileSettingsChanged).toBe(true);
      });
      it('if "profile.generalInfo.englishLevel" was changed', async () => {
        const event = 'b2+';
        const path = 'generalInfo.englishLevel';
        wrapper.setState(state);
        await instance.onProfileSettingsChange(event, path);
        expect((wrapper.state() as any).profile.generalInfo.englishLevel).toBe('b2+');
      });
      it('if field added to "profile.generalInfo.educationHistory"', async () => {
        const event = {
          type: 'add',
        };
        const path = 'generalInfo.educationHistory';
        wrapper.setState(state);
        await instance.onProfileSettingsChange(event, path);
        expect((wrapper.state() as any).profile.generalInfo.educationHistory).toEqual([
          {
            graduationYear: '2019',
            faculty: 'TT',
            university: 'Test',
          },
          {
            graduationYear: null,
            faculty: null,
            university: null,
          },
        ]);
        expect((wrapper.state() as any).isInitialProfileSettingsChanged).toBe(true);
      });
      it('if field deleted from "profile.generalInfo.educationHistory"', async () => {
        const event = {
          type: 'delete',
          index: 0,
        };
        const path = 'generalInfo.educationHistory';
        wrapper.setState(state);
        await instance.onProfileSettingsChange(event, path);
        expect((wrapper.state() as any).profile.generalInfo.educationHistory).toEqual([]);
      });
      it('if some other field was changed', async () => {
        const event = {
          target: {
            value: 'Hello everyone, my name is Mike.',
          },
        };
        const path = 'generalInfo.aboutMyself';
        wrapper.setState(state);
        await instance.onProfileSettingsChange(event, path);
        expect((wrapper.state() as any).profile.generalInfo.aboutMyself).toEqual('Hello everyone, my name is Mike.');
        expect((wrapper.state() as any).isInitialProfileSettingsChanged).toBe(true);
      });
    });
  });
  describe('changeProfilePageMode', () => {
    describe('Should set state correctly', () => {
      it('if mode = "edit" was passed', async () => {
        const mode = 'edit';
        wrapper.setState({ ...state, isEditingModeEnabled: false });
        expect((wrapper.state() as any).isEditingModeEnabled).toBe(false);
        await instance.changeProfilePageMode(mode);
        expect((wrapper.state() as any).isEditingModeEnabled).toBe(true);
      });
      it('if mode = "view" was passed', async () => {
        const mode = 'view';
        wrapper.setState({ ...state, isEditingModeEnabled: true });
        expect((wrapper.state() as any).isEditingModeEnabled).toBe(true);
        await instance.changeProfilePageMode(mode);
        expect((wrapper.state() as any).isEditingModeEnabled).toBe(false);
      });
    });
  });
  describe('saveProfile', () => {
    it('Should set state correctly', async () => {
      const saveProfileMock = jest
        .spyOn(instance['userService'], 'saveProfileInfo')
        .mockImplementation(() => Promise.resolve());

      const profile = {
        generalInfo: {
          aboutMyself: 'Hello',
          educationHistory: [
            {
              graduationYear: '2019',
              faculty: 'TT',
              university: 'Test',
            },
          ],
          englishLevel: 'c1',
          locationId: 778,
          locationName: 'Hrodna',
        },
        contacts: {
          telegram: 'test',
        },
        permissionsSettings: {
          isProfileVisible: { all: true },
          isAboutVisible: { mentor: true, student: false, all: false },
          isEducationVisible: { mentor: true, student: false, all: false },
          isEnglishVisible: { student: true, all: true },
          isEmailVisible: { student: true, all: true },
          isTelegramVisible: { student: true, all: true },
          isSkypeVisible: { student: true, all: false },
          isPhoneVisible: { student: true, all: false },
          isContactsNotesVisible: { student: true, all: false },
          isLinkedInVisible: { mentor: true, student: false, all: false },
          isPublicFeedbackVisible: { mentor: true, student: true, all: false },
          isMentorStatsVisible: { mentor: true, student: true, all: false },
          isStudentStatsVisible: { student: false, all: false },
        },
      };
      wrapper.setState({
        ...state,
        profile,
        isInitialPermissionsSettingsChanged: true,
        isInitialProfileSettingsChanged: true,
      });

      await instance.saveProfile();
      expect((wrapper.state() as any).isSaving).toBe(false);
      expect((wrapper.state() as any).isInitialPermissionsSettingsChanged).toBe(false);
      expect((wrapper.state() as any).isInitialProfileSettingsChanged).toBe(false);
      expect((wrapper.state() as any).initialPermissionsSettings).toEqual(profile.permissionsSettings);
      expect((wrapper.state() as any).initialProfileSettings).toEqual(profile);
      saveProfileMock.mockReset();
    });
  });
  describe('hadStudentCoreJSInterview', () => {
    describe('Should return', () => {
      it('"true" if student has an "interviewFormAnswers" in one of the task', () => {
        const studentStats = [
          {
            courseFullName: 'test',
            courseName: 'test',
            locationName: 'Minsk',
            tasks: [
              {},
              {
                interviewFormAnswers: {},
              },
              {},
              {},
            ],
          },
        ];
        const result = instance.hadStudentCoreJSInterview(studentStats);
        expect(result).toBe(true);
      });
      it('"false" if student has not an "interviewFormAnswers" in one of the task', () => {
        const studentStats = [
          {
            courseFullName: 'test',
            courseName: 'test',
            locationName: 'Minsk',
            tasks: [{}, {}, {}],
          },
        ];
        const result = instance.hadStudentCoreJSInterview(studentStats);
        expect(result).toBe(false);
      });
    });
  });
  describe('getStudentCoreJSInterviews', () => {
    it('Should return info about CoreJS interviews', () => {
      const studentStats = [
        {
          courseFullName: 'test',
          courseName: 'test',
          locationName: 'Minsk',
          tasks: [
            {},
            {},
            {
              interviewer: {
                name: 'Dima Petrov',
                githubId: 'dip',
              },
              comment: 'Test',
              score: 9,
              interviewDate: '1652094809232',
              interviewFormAnswers: {},
              name: 'CoreJS Interview',
            },
            {},
          ],
        },
      ];
      const result = instance.getStudentCoreJSInterviews(studentStats);
      expect(result).toEqual([
        {
          courseFullName: 'test',
          courseName: 'test',
          interviews: [
            {
              answers: {},
              interviewer: {
                name: 'Dima Petrov',
                githubId: 'dip',
              },
              comment: 'Test',
              score: 9,
              interviewDate: '1652094809232',
              name: 'CoreJS Interview',
            },
          ],
          locationName: 'Minsk',
        },
      ]);
    });
  });
});
