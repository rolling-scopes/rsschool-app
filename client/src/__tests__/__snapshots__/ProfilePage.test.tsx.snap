// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage Should render correctly if full profile info is in the state 1`] = `
<Fragment>
  <LoadingScreen
    show={true}
  >
    <Header
      isProfileEditingModeEnabled={false}
      isProfilePage={false}
      isSaveButtonVisible={false}
      onChangeProfilePageMode={[Function]}
      onSaveClick={[Function]}
      username="mikhama"
    />
    <Spin
      delay={200}
      size="default"
      spinning={false}
      wrapperClassName=""
    >
      <div
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <Masonry
          breakpointCols={
            Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            }
          }
          className="jsx-3803498300"
          columnClassName="jsx-110195169"
        >
          <div
            key="card-0"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <MainCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": Array [
                    Object {
                      "faculty": "TT",
                      "graduationYear": "2019",
                      "university": "Test",
                    },
                  ],
                  "englishLevel": "a2+",
                  "location": Object {
                    "cityName": "Minsk",
                    "countryName": "Belarus",
                  },
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-1"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <AboutCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": Array [
                    Object {
                      "faculty": "TT",
                      "graduationYear": "2019",
                      "university": "Test",
                    },
                  ],
                  "englishLevel": "a2+",
                  "location": Object {
                    "cityName": "Minsk",
                    "countryName": "Belarus",
                  },
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-2"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <EnglishCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": Array [
                    Object {
                      "faculty": "TT",
                      "graduationYear": "2019",
                      "university": "Test",
                    },
                  ],
                  "englishLevel": "a2+",
                  "location": Object {
                    "cityName": "Minsk",
                    "countryName": "Belarus",
                  },
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-3"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <EducationCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": Array [
                    Object {
                      "faculty": "TT",
                      "graduationYear": "2019",
                      "university": "Test",
                    },
                  ],
                  "englishLevel": "a2+",
                  "location": Object {
                    "cityName": "Minsk",
                    "countryName": "Belarus",
                  },
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-4"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <ContactsCard
              data={Object {}}
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-5"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <PublicFeedbackCard
              data={
                Array [
                  Object {},
                ]
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-6"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <StudentStatsCard
              data={
                Array [
                  Object {
                    "courseFullName": "test",
                    "courseName": "test",
                    "locationName": "Minsk",
                    "tasks": Array [
                      Object {
                        "interviewFormAnswers": Object {},
                      },
                    ],
                  },
                ]
              }
              isEditingModeEnabled={false}
              isProfileOwner={false}
              onPermissionsSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
              username="mikhama"
            />
          </div>
          <div
            key="card-7"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <MentorStatsCard
              data={
                Array [
                  Object {},
                ]
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-8"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <CoreJSIviewsCard
              data={
                Array [
                  Object {
                    "courseFullName": "test",
                    "courseName": "test",
                    "interview": Object {
                      "answers": Object {},
                      "comment": undefined,
                      "interviewer": undefined,
                      "score": undefined,
                    },
                    "locationName": "Minsk",
                  },
                ]
              }
            />
          </div>
          <div
            key="card-9"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <PreScreeningIviewsCard
              data={
                Array [
                  Object {},
                ]
              }
            />
          </div>
        </Masonry>
        <JSXStyle
          id="3803498300"
        >
          div.jsx-3803498300{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-left:-16px;width:auto;}
        </JSXStyle>
        <JSXStyle
          id="110195169"
        >
          div.jsx-110195169{padding-left:16px;background-clip:padding-box;}
        </JSXStyle>
      </div>
    </Spin>
  </LoadingScreen>
</Fragment>
`;
